snapcraft/internal/lxd.py

```diff
@@ -204,18 +218,83 @@ def _ensure_container(self):
     def _setup_project(self):
         self._ensure_mount(self._project_folder, self._source)
 
+    def _get_container_address(self):
+        network = self._get_container_status()['state']['network']['eth0']
+        for address in network['addresses']:
+            if address['family'] == 'inet':
+                return address['address']
+        raise RuntimeError('No IP found for {}'.format(self._container_name))
+
     def _ensure_mount(self, destination, source):
         logger.info('Mounting {} into container'.format(source))
+        remote, container_name = self._container_name.split(':')
+        if remote != 'local':
+            self._remote_mount(destination, source)
+        else:
+            devices = self._get_container_status()['devices']
+            if destination not in devices:
+                check_call([
+                    'lxc', 'config', 'device', 'add', self._container_name,
+                    destination, 'disk', 'source={}'.format(source),
+                    'path=/{}'.format(destination)])
+
+    def _remote_mount(self, destination, source):
+        # Remove project folder in case it was used "locally" before
         devices = self._get_container_status()['devices']
-        if destination not in devices:
+        if destination in devices:
             check_call([
-                'lxc', 'config', 'device', 'add', self._container_name,
+                'lxc', 'config', 'device', 'remove', self._container_name,
                 destination, 'disk', 'source={}'.format(source),
                 'path=/{}'.format(destination)])
 
+        # Generate an SSH key and add it to the container's known keys
```

Comment:
The diff isn't being helpful here: where would this keyfile end up? Would it be useful to default to `~/.ssh/id_rsa.pub` if it already exists rather than dumping random keys on disk?